docs/sdk/src/guides/enable_elastic_scaling_mvp.rs

```diff
@@ -0,0 +1,142 @@
+//!    capped at 2 seconds of execution. Therefore, assuming the full 2 seconds are used, a
+//!    parachain can only utilise at most 3 cores in a relay chain slot of 6 seconds. If the full
+//!    execution time is not being used, higher core counts can be achieved.
+//! 2. **Single collator requirement for consistently scaling beyond a core at full authorship
+//!    duration of 2 seconds per block.** Using the current implementation with multiple collators
+//!    adds additional latency to the block production pipeline. Assuming block execution takes
+//!    about the same as authorship, the additional overhead is equal the duration of the authorship
+//!    plus the block announcement. Since each collator must first import the previous block before
+//!    authoring a new one, this would amount to 12 seconds for 3 cores and 8 seconds for 2 cores.
+//! 3. **Trusted collator set.** The collator set needs to be trusted until there’s a mitigation
+//!    that would prevent or deter multiple collators from submitting the same collation.
+//!    A solution is being discussed [here](https://github.com/polkadot-fellows/RFCs/issues/92).
+//! 4. **Fixed scaling.** For true elasticity, the parachain must be able to seamlessly acquire or
+//!    sell coretime as the user demand grows and shrinks over time, in an automated manner. This is
+//!    currently lacking - a parachain can only scale up or down by “manually” acquiring coretime.
+//!    This is not in the scope of the relay chain functionality. Parachains can already start
+//!    implementing such autoscaling, but we aim to provide a framework/examples for developing
+//!    autoscaling strategies.
+//!
+//! ## Using elastic scaling MVP
+//!
+//! ### Prerequisites
+//!
+//! - Ensure Asynchronous Backing is enabled on the network and you have enabled it on the parachain
+//!   using [this guide](https://wiki.polkadot.network/docs/maintain-guides-async-backing).
+//! - Ensure the `AsyncBackingParams.max_candidate_depth` value is configured to a value that is at
+//!   least double the maximum targeted parachain velocity. For example, if the parachain will build
+//!   at most 3 candidates per relay chain block, the `max_candidate_depth` should be at least 6.
+//! - Use a trusted single collator for maximum throughput.
+//! - Ensure enough coretime is assigned to the parachain. For maximum throughput the upper bound is
+//!   3 cores.
+//! - Use the latest cumulus release, which includes the necessary elastic scaling changes.
+//!
+//! The following steps assume using the cumulus parachain template.
+//!
+//! ### Phase 1 - Update Parachain Node
+//!
+//! This phase consists of plugging in the new slot-based collator node.
+//!
+//! 1. In `node/src/service.rs` import the slot based collator instead of the lookahead collator.
+//!
+//! ```rust
+//! use cumulus_client_consensus_aura::collators::slot_based::{self as aura, Params as AuraParams};
+//! ```
+//!
+//! 2. In `start_consensus()`
```

Comment:
This should use `docify` into `parachain-template`. Possibly, it will be hard to do it, and I belive this is a good thing, as it will force us to simplify the API a bit. Give it a try please. As noted above, it is similar to the suggestions in https://github.com/paritytech/polkadot-sdk/pull/4363#issuecomment-2141171928

Reply:
For example, my worry with this doc rn is that the moment we change anything in the parachain template, it will likely become invalid.

Reply:
What you're saying makes sense, but at least part of this guide is for parachains that are already running and want to update. The default template will already be elastic-scaling ready, much like it is already async backing ready.
Still, the document for updating to async backing was targeted at existing parachains (the ones that start from the template barely need to do anything).

Reply:
maybe it would be best to first update the template and then just reference it in the doc as an example, instead of giving find and replace instructions. however, the feature is not yet stable enough so that we can update the template. therefore I'll leave the PR as it is for now (that's also why it's a draft)

Reply:
so basically there are two conlicting things:
1. the document should help existing parachains upgrade
2. keeping the document up to date with the template

If we want 2, the document will lack all the changes needed for 1